### ISSUE TITLE
How to support key_padding_mask

### ISSUE THREAD

**Issue by YJYJLee** [facebookresearch/xformers#825]
# ❓ Questions and Help
Hi, I am trying to replace existing `torch.nn.functional.scaled_dot_product_attention` with `xformers.ops.memory_efficient_attention`.

I want to use `xformers.ops.fmha.flash` and `xformers.ops.fmha.triton` as ops, but I am struggling with how to handle **key_padding_masks** in the original code.
I see that 
   - `xformers.ops.fmha.flash` supports `Lower Triangular Mask`, `BlockDiagonalMask` and `BlockDiagonalCausalMask`
   - `xformers.ops.fmha.triton` supports `Lower Triangular Mask`

Question is, 
   - When using `xformers.ops.fmha.flash`, what is the most efficient way to apply key_padding_mask here? I referred to this [link](https://facebookresearch.github.io/xformers/components/ops.html#xformers.ops.fmha.attn_bias.BlockDiagonalMask) and tried using `BlockDiagonalMask`, but it requires a lot of processing (actually leads to significant slowdown) when you already have queries, values, keys of shape [B, H, M, K] after projection. What is the most efficient way to use BlockDiagonalMask in this case? I attached the code below I used.
     ```
     shape_list = torch.sum(key_padding_mask==0, -1)       # key_padding_mask.shape=[B, H, 1. K], this code counts valid key tokens
     new_q = [q[i].unsqueeze(0).unsqueeze(-2) for i in range(q.shape[0])]
     new_k = [k[i,:shape_list[i], :].unsqueeze(0).unsqueeze(-2) for i in range(q.shape[0])]
     new_v = [v[i,:shape_list[i], :].unsqueeze(0).unsqueeze(-2) for i in range(q.shape[0])]
     new_attn_mask, new_q, new_k, new_v = fmha.attn_bias.BlockDiagonalMask.from_tensor_lists_qkv(new_q, new_k, new_v)
      attn = memory_efficient_attention(new_q, new_k, new_v, attn_bias=new_attn_mask, p=dropout_p, op=(FwOp, BwOp))
     ```
     This code seems to be working as same as original `torch.nn.functional.scaled_dot_product_attention` with key_padding_mask, but I think it is super inefficient. Could you provide some example of how to prepare query, key, value and mask in this case?
   - When using `xformers.ops.fmha.triton` , I can't come up with any workarounds on how to handle key_padding_masks since `BlockDiagonalMask` is not available for this implementation. Is there any way to do this?
   
  Thanks for the help in advance!
